File: src/main/java/mil/nga/tiff/field/type/enumeration/ResolutionUnit.java
package mil.nga.tiff.util;

import java.util.Arrays;
import java.util.Optional;

public enum ResolutionUnit {
    NO(1),
    INCH(2),
    CENTIMETER(3);

    private final int id;

    ResolutionUnit(int id) {
        this.id = id;
    }

    public static ResolutionUnit findById(int id) {
        return Arrays.stream(values())
            .filter(o -> o.id == id)
            .findAny()
            .orElseThrow(() -> new TiffException("Invalid ID: " + id));
    }

    public int getId() {
        return id;
    }
}

File: src/main/java/mil/nga/tiff/field/type/enumeration/ResolutionUnit.kt
package mil.nga.tiff.field.type.enumeration

enum class ResolutionUnit(val id: Int) {
    NO(1),
    INCH(2),
    CENTIMETER(3);

    companion object {
        @JvmStatic
        fun findById(id: Int): ResolutionUnit {
            return entries.first { it.id == id}
        }
    }
}
